tests/integration/codexclient.nim

```diff
@@ -138,27 +138,68 @@ proc getSlots*(client: CodexClient): ?!seq[Slot] =
 
 proc postAvailability*(
     client: CodexClient,
-    size, duration, minPrice, maxCollateral: UInt256
+    totalSize, duration, minPrice, maxCollateral: UInt256
 ): ?!Availability =
   ## Post sales availability endpoint
   ##
   let url = client.baseurl & "/sales/availability"
   let json = %*{
-    "size": size,
+    "totalSize": totalSize,
     "duration": duration,
     "minPrice": minPrice,
     "maxCollateral": maxCollateral,
   }
   let response = client.http.post(url, $json)
-  doAssert response.status == "200 OK", "expected 200 OK, got " & response.status & ", body: " & response.body
+  doAssert response.status == "201 Created", "expected 201 Created, got " & response.status & ", body: " & response.body
   Availability.fromJson(response.body.parseJson)
 
+proc patchAvailabilityRaw*(
+    client: CodexClient,
+    availabilityId: AvailabilityId,
+    totalSize, freeSize, duration, minPrice, maxCollateral: ?UInt256 = UInt256.none
+): Response =
+  ## Updates availability
+  ##
+  let url = client.baseurl & "/sales/availability/" & $availabilityId
+  var json = %*{}
+
+  if totalSize =? totalSize:
+    json["totalSize"] = %totalSize
+
+  if freeSize =? freeSize:
+    json["freeSize"] = %freeSize
+
+  if duration =? duration:
+    json["duration"] = %duration
+
+  if minPrice =? minPrice:
+    json["minPrice"] = %minPrice
+
+  if maxCollateral =? maxCollateral:
+    json["maxCollateral"] = %maxCollateral
+
+  client.http.patch(url, $json)
```

Comment:
Serializing an optionalized type will output `null` for `none` fields, which are handled correctly in deserialization on the api side.
```suggestion
type OptRestAvailability = Optionalize(RestAvailability)
let availability = OptRestAvailability(totalSize: totalSize, freeSize: freeSize, duration: duration, minPrice: minPrice, maxCollateral: maxCollateral)

client.http.patch(url, availability.toJson)
```

Another thought: we could also consider exporting this type from `rest/json` instead of creating it each time.

Reply:
I was banging my head on why integration tests were not passing and it seemed like there were no changes updated on the node's side, while when I tested it locally it worked. It turned out that yours suggestion is actually serializing into empty JSON object `{}` 😳

It is because the `Optionalize` does not preserve the `serialize` pragmas. Honestly, I don't want to tinker with the macro anymore as that would be another time blackhole, so I will reject your proposal, even though I like it more than the original. 

If/when we find the need to extend the `Optionalize` to honor `serialize` pragmas then we can revisit this part.

Reply:
No problem, I understand. For future reference, you can annotate the type with `{.serialize.}` and then there's no need to annotate each property, ie all properties will be serialized by default.